### PR TITLE
Incorporated (Jimmie's) build and submit-job notebooks, and a sample pge notebook

### PR DESCRIPTION
**Pull request by alexdunncs**
These are now instantiated and automatically populated with the project's name upon `create` , and provide useful example documentation (there's still a bunch of manual edits necessary, but it's an improvement).

I will refine these as I continue working through the SOAMC UX improvement work. 

Hopefully these changes are acceptable for incorporation - if not, let me know what you'd like to see.

N.B. This updates papermill (which at the time I thought was a necessary part of a bug I'd been chasing but probably isn't).  This can be reverted if undesirable.

All tests pass

```
/home/parallels/dev/soamc/notebook_pge_wrapper/venv/bin/python /snap/pycharm-professional/237/plugins/python/helpers/pycharm/_jb_unittest_runner.py --path /home/parallels/dev/soamc/notebook_pge_wrapper/test
Testing started at 4:36 PM ...
Launching unittests with arguments python -m unittest discover -s /home/parallels/dev/soamc/notebook_pge_wrapper/test -t /home/parallels/dev/soamc/notebook_pge_wrapper in /home/parallels/dev/soamc/notebook_pge_wrapper

2021-03-31 16:36:22 [INFO] Input Notebook:  test/notebook_pges/test.ipynb
2021-03-31 16:36:22 [INFO] Input Notebook:  test/notebook_pges/test.ipynb
2021-03-31 16:36:22 [INFO] Input Notebook:  test/notebook_pges/test.ipynb
2021-03-31 16:36:22 [INFO] Input Notebook:  test/notebook_pges/test2.ipynb
2021-03-31 16:36:22 [INFO] Input Notebook:  test/notebook_pges/test.ipynb


Ran 7 tests in 2.066s

OK

Process finished with exit code 0
2021-03-31 16:36:22 [INFO] Input Notebook:  /home/parallels/dev/soamc/notebook_pge_wrapper/test/notebook_pges/test.ipynb
2021-03-31 16:36:22 [INFO] Input Notebook:  /home/parallels/dev/soamc/notebook_pge_wrapper/test/notebook_pges/test.ipynb
2021-03-31 16:36:22 [INFO] Output Notebook: test-output.ipynb
2021-03-31 16:36:22 [INFO] Generating grammar tables from /home/parallels/dev/soamc/notebook_pge_wrapper/venv/lib/python3.8/site-packages/blib2to3/Grammar.txt
2021-03-31 16:36:22 [INFO] Writing grammar tables to /home/parallels/.cache/black/20.8b1/Grammar3.8.5.final.0.pickle
2021-03-31 16:36:22 [INFO] Writing failed: [Errno 2] No such file or directory: '/home/parallels/.cache/black/20.8b1/tmp8rk5a9y5'
2021-03-31 16:36:22 [INFO] Generating grammar tables from /home/parallels/dev/soamc/notebook_pge_wrapper/venv/lib/python3.8/site-packages/blib2to3/PatternGrammar.txt
2021-03-31 16:36:22 [INFO] Writing grammar tables to /home/parallels/.cache/black/20.8b1/PatternGrammar3.8.5.final.0.pickle
2021-03-31 16:36:22 [INFO] Writing failed: [Errno 2] No such file or directory: '/home/parallels/.cache/black/20.8b1/tmp2do3_03m'
Executing:   0%|          | 0/3 [00:00<?, ?cell/s]/home/parallels/dev/soamc/notebook_pge_wrapper/venv/lib/python3.8/site-packages/traitlets/config/configurable.py:82: DeprecationWarning: Passing unrecognized arguments to super(PapermillNotebookClient).__init__(input_path='/home/parallels/dev/soamc/notebook_pge_wrapper/test/notebook_pges/test.ipynb').
object.__init__() takes exactly one argument (the instance to initialize)
This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
2021-03-31 16:36:24 [INFO] Executing notebook with kernel: python3
2021-03-31 16:36:24 [INFO] Executing Cell 1---------------------------------------
2021-03-31 16:36:24 [INFO] Ending Cell 1------------------------------------------
2021-03-31 16:36:24 [INFO] Executing Cell 2---------------------------------------
Executing:  33%|███▎      | 1/3 [00:01<00:03,  1.62s/cell]2021-03-31 16:36:24 [INFO] Ending Cell 2------------------------------------------
2021-03-31 16:36:24 [INFO] Executing Cell 3---------------------------------------
2021-03-31 16:36:24 [INFO] hello world
a: 1234567 type:  <class 'int'>
b: hello this is my test value for (b) type:  <class 'str'>
c: 123.123 type:  <class 'float'>
d: [9, 8, 7, 6, 5, 5, 4, 3, 2, 1] type:  <class 'list'>
e: ['a', 'b', 'c'] type:  <class 'list'>
f: test value for f type:  <class 'str'>
g: ['c', 'b', 'a'] type:  <class 'list'>

2021-03-31 16:36:24 [INFO] Ending Cell 3------------------------------------------
Executing: 100%|██████████| 3/3 [00:01<00:00,  1.62cell/s]

```

### CODE REVIEW

**Comment by DustinKLo** [comment]
@alexdunncs  i'm not too aware of the changes made here

can you demo this sometime to @mkarim2017 @jimmie @pymonger and I so we can get a better picture of whats going on
and can you link a ticket also?

**Comment by alexdunncs** [comment]
@DustinKLo sure can

The changes here I think we talked about a meeting or two ago which we thought might be useful, and amount to "copy in some useful extra templated stuff when you run `notebook-pge-wrapper create`"

https://github.com/alexdunncs/notebook_pge_wrapper/blob/develop/templates/pge_create.ipynb
https://github.com/alexdunncs/notebook_pge_wrapper/blob/develop/templates/sample_pge.ipynb
https://github.com/alexdunncs/notebook_pge_wrapper/blob/develop/templates/submit_job.ipynb

Happy to demo it, but that's all there is to it.

It's most-related to https://hysds-core.atlassian.net/browse/SOAMC-75

Let me know how you'd like me to proceed.

**Comment by DustinKLo** [comment]
should we move `sample_pge.ipynb` out of the `notebook_pges` directory on creation?
so that spec generation and `container-builder` wont register a empty job by accident

File: notebook_pge_wrapper/cli.py [hunk 1]
```diff
@@ -56,9 +64,26 @@ def create(project):
...
 
+    # create sample pge notebook
+    copyfile(sample_pge_notebook_file, os.path.join(notebook_pges_directory, __SAMPLE_PGE_NOTEBOOK_FILE))
+
     # create README.md
     copyfile(readme_file, os.path.join(project, __README_FILE))
 
+    # create pge_create notebook
+    pge_create_notebook_dest = os.path.join(project, __PGE_CREATE_NOTEBOOK_FILE)
+    with open(pge_create_notebook_file, 'r') as pge_create_infile, \
```
Comment (DustinKLo) [inline notebook_pge_wrapper/cli.py]: we can rename `pge_create_infile` to `fin` and `pge_create_outfile` to `fout` for readability

File: notebook_pge_wrapper/cli.py [hunk 2]
```diff
@@ -56,9 +64,26 @@ def create(project):
...
+            open(pge_create_notebook_dest, 'w+') as pge_create_outfile:
+        templated_pge_create_content = pge_create_infile.read().replace('PGE_NAME_PLACEHOLDER', project)
+        pge_create_outfile.write(templated_pge_create_content)
+
+    # create submit_job notebook
+    submit_job_notebook_dest = os.path.join(project, __SUBMIT_JOB_NOTEBOOK_FILE)
+    with open(submit_job_notebook_file, 'r') as submit_job_infile, \
```
Comment (DustinKLo) [inline notebook_pge_wrapper/cli.py]: rename `submit_job_infile` to `fin` and `submit_job_outfile` to `fout` for readability

File: requirements.txt [hunk 3]
```diff
@@ -1,2 +1,2 @@
 click>=7.1.2
-papermill==2.2.0
+papermill==2.3.3
```
Comment (DustinKLo) [inline requirements.txt]: should be safe to remove the version lock
can use `>=2.3.3` instead


unless you encountered a bug with later versions of `papermill`

**Review by DustinKLo — CHANGES_REQUESTED** [review]
can you remove the trailing empty cells in:

- `pge_create.ipynb`
- `sample_pge.ipynb`
- `submit_job.ipynb`

@jimmie @mkarim2017  can you test these changes when you have time
if its intuitive enough for you then its good for me 👍

File: requirements.txt (hunk 3 above)
Comment (alexdunncs) [reply requirements.txt]: I've been bitten multiple times by supposedly-compatible updates on other projects so lock versions by default, but will make this update (and the others).

**Comment by alexdunncs** [comment]
> should we move `sample_pge.ipynb` out of the `notebook_pges` directory on creation?
> so that spec generation and `container-builder` wont register a empty job by accident

@DustinKLo My intent was to provide a skeleton that could be immediately built and run.  If this isn't appropriate, we can put it in an examples directory or something.

**Comment by DustinKLo** [comment]
> > should we move `sample_pge.ipynb` out of the `notebook_pges` directory on creation?
> > so that spec generation and `container-builder` wont register a empty job by accident
> 
> @DustinKLo My intent was to provide a skeleton that could be immediately built and run. If this isn't appropriate, we can put it in an examples directory or something.

@jimmie @mkarim2017 i can use your input on this, where would you want the sample `.ipynb` file to be
we can leave it in `notebook_pges/` but PGEs might accidentally get built
we can have a `example/` directory and add it there
i'm up for suggestions

**Review by jimmie — DISMISSED** [review]
LGTM.

**Review by DustinKLo — DISMISSED** [review]
LGTM 👍 
feel free to merge @alexdunncs

**Review by DustinKLo — COMMENTED** [review]
my mistake i didn't see the additional commits after jimmy's approval
let us know if its ready for another review @alexdunncs

**Comment by alexdunncs** [comment]
@DustinKLo should be good as-is, though I should(?) incorporate some better explanation of the templated notebooks in the main readme soon.

If someone will review (and maybe just double-check with a `notebook-pge-wrapper create whateverName`, though I've done it myself), I'll merge.

Automated tests pass

File: notebook_pge_wrapper/spec_generator.py [hunk 4]
```diff
@@ -215,7 +215,7 @@ def generate_job_spec(time_limit=__DEFAULT_TIME_LIMIT, soft_time_limit=__DEFAULT
     """
     example: {
         "required_queues":["system-jobs-queue"],
-        "command":"/path/to/papermill_wrapper.py $HOME/hello_world.ipynb",
+        "command":"/path/to/papermill_wrapper.py $HOME/PGE_NAME_PLACEHOLDER.ipynb",
```
Comment (DustinKLo) [inline notebook_pge_wrapper/spec_generator.py:218]: i should make the `PGE_NAME_PLACEHOLDER` auto-fill in my PR #11

**Review by DustinKLo — APPROVED** [review]
LGTM 👍 
feel free to merge @alexdunncs

**Review by pymonger — APPROVED** [review]
LGTM

**Pull request merged**